rskj-core/src/main/java/org/ethereum/core/TransactionPoolAddResult.java

```diff
@@ -42,11 +48,15 @@ public void ifTransactionWasNotAdded(Consumer<String> errorConsumer) {
         }
     }
 
-    public static TransactionPoolAddResult ok() {
-        return new TransactionPoolAddResult(true, null);
+    public static TransactionPoolAddResult ok(Transaction transaction) {
+        return new TransactionPoolAddResult(true, null, Collections.singletonList(transaction));
     }
 
     public static TransactionPoolAddResult withError(String errorMessage) {
-        return new TransactionPoolAddResult(false, errorMessage);
+        return new TransactionPoolAddResult(false, errorMessage, null);
```

Comment:
This could cause a NPE in the check "if(!result.getTransactionsAdded().isEmpty()) {" L53 TransactionGateway.

Reply:
fixed